pkg/controller/secrets_test.go

```diff
@@ -0,0 +1,193 @@
+// Copyright (C) 2020, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package controller
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/verrazzano/verrazzano-operator/pkg/constants"
+	. "github.com/verrazzano/verrazzano-operator/pkg/managed"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test creating and getting a secret
+func TestKubeSecrets_CreateGet(t *testing.T) {
+	connections := GetManagedClusterConnections()
+	connection := connections["cluster1"]
+
+	origSecret := corev1.Secret{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: "test-secret-namespace-1",
+			Name:      "test-secret-name-1"},
+		Immutable:  nil,
+		Data:       nil,
+		StringData: nil,
+		Type:       "test-secret-type",
+	}
+
+	secrets := KubeSecrets{
+		namespace:     "test-secret-namespace-1",
+		kubeClientSet: connection.KubeClient,
+		secretLister:  connection.SecretLister,
+	}
+
+	secret, err := secrets.Create(&origSecret)
+	assert.NoErrorf(t, err, "Can't create secret: %v", err)
```

Comment:
Just FYI, use can use omit the 't' param  and use assert(err, ..   if you do this at the start of the function.  I learned this from Mike

	assert := assert.New(t)

Reply:
Nice, much better.

Reply:
Fixed.